File: 2870-minimum-number-of-operations-to-make-array-empty/2870-minimum-number-of-operations-to-make-array-empty.java
class Solution {
    public int minOperations(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        int n = nums.length;
        int result = 0;
        for(int i=0; i<n; i++){
            map.put(nums[i], map.getOrDefault(nums[i],0)+1);
        }
        for(Map.Entry<Integer, Integer> entry: map.entrySet()){
            int freq = entry.getValue();
            
            if(freq == 1){
                return -1;
            }
            result = result + (int)Math.ceil((double)freq/3);
        }
        return result;
    }
}

File: 2870-minimum-number-of-operations-to-make-array-empty/2870-minimum-number-of-operations-to-make-array-empty.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        unordered_map<int, int> map;
        int n = nums.size();
        int result = 0;
        for(int i=0; i<n; i++){
            map[nums[i]]++;
        }
        for(auto &element: map){
            int freq = element.second;
            if(freq == 1){
                return -1;
            }
            result = result + ceil((double)freq/3);
        }
        return result;
    }
};